### PR TITLE
Splitting out and Parser for the SEXP module

### PR DESCRIPTION
**Pull request by mariari**
Merge after #691 

This creates a parser for sexps

### CODE REVIEW

File: library/StandardLibrary/src/Juvix/Library/Sexp/Parser.hs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  digits <- Atto.takeWhile Lexer.digit
+  case Char8.readInteger digits of
+    Just (x, _) -> pure x
+    Nothing -> fail "didn't parse an int"
+
+--------------------------------------------------------------------------------
+-- Helpers taken from the other parser
+--------------------------------------------------------------------------------
+
+-- eddited a bit from the other parser
```
Comment (cwgoes) [inline library/StandardLibrary/src/Juvix/Library/Sexp/Parser.hs:66]:
```suggestion
-- edited a bit from the other parser
```

File: library/StandardLibrary/src/Juvix/Library/Sexp/Parser.hs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  s <-
+    Atto.takeWhile1
+      ( \x ->
+          Lexer.validStartSymbol x
+            || Lexer.validMiddleSymbol x
+            || Lexer.validInfixSymbol x
+      )
+  Encoding.decodeUtf8 s |> internText |> NameSymbol.fromSymbol |> pure
+
+-- Code stolen from the other paser ☹
```
Comment (cwgoes) [inline library/StandardLibrary/src/Juvix/Library/Sexp/Parser.hs:53]:
```suggestion
-- Code stolen from the other parser ☹
```

**Review by cwgoes — APPROVED** [review]
ACK modulo nits, do we also want tests for the parser itself?

**Comment by mariari** [comment]
> ACK modulo nits, do we also want tests for the parser itself?

We should, those passes I made were sort of implicit tests. I'll make some tests first then monday and then merge this

**Pull request merged**